src/features/blocks-list/index.tsx

```diff
@@ -2,29 +2,40 @@ import React, { useMemo } from 'react';
 import { color, Flex, FlexProps, Grid, Spinner } from '@stacks/ui';
 import { Section } from '@components/section';
 import { HoverableItem } from '@components/hoverable';
-import { useBlocksList } from './hooks';
 import { Caption } from '@components/typography';
 import { BlockItem } from './block-list-item';
 import { MicroblockItem } from './microblock-list-item';
 import { SafeSuspense } from '@components/ssr-safe-suspense';
 import { SectionFooterAction } from '@components/section-footer-button';
 import { DEFAULT_LIST_LIMIT } from '@common/constants';
+import { useInfiniteQuery } from 'react-query';
+import { getNextPageParam } from '@store/common';
+import { useHomeQueries } from '@features/home/useHomeQueries';
+
+function useBlockList(limit: number) {
+  const queries = useHomeQueries();
+  const { data: blocks, ...actions } = useInfiniteQuery(
+    ['blocks'],
+    ({ pageParam }) => queries.fetchBlocks(limit, pageParam || 0)(),
+    { getNextPageParam }
+  );
+  return { blocks, actions };
```

Comment:
Consider returning a useMemo here. If you use `useBlockList`s in a dependency array now, it'll reevaluate on every render, which can be hard to track down when it crashes the app.